paladin/mod.ts

```diff
@@ -0,0 +1,203 @@
+    report_uri?: string;
+  };
+  "X-DNS-Prefetch-Control"?: boolean;
+  "Content-Security-Policy"?: string;
+}
+
+/**
+ * A logger middleware inspired by https://www.npmjs.com/package/morgan.
+ *
+ * @param configs - See IDexterConfigs
+ */
+export function Paladin(
+  configs: Configs = {},
+) {
+
+  if (!configs.hsts) {
+    configs.hsts = {};
+  }
+  if (!configs.expect_ct) {
+    configs.expect_ct = {};
+  }
+
+  // Default configs when no `configs` param is passed in
+  const defaultConfigs = {
+    "X-XSS-Protection": "1; mode=block",
+    "X-Content-Type-Options": "nosniff",
+    hsts: {
+      max_age: 5184000, // 60 days
+      include_sub_domains: "include_sub_domains",
+    },
+    "X-Powered-By": false, // False for removing the header
+    "X-Frame-Options": "SAMEORIGIN",
+    "X-DNS-Prefetch-Control": false,
+  };
+
+  /**
+   * The middleware function that's called by Drash.
+   *
+   * @param request - The request object.
+   * @param response - (optional) The response object.
+   */
+  function paladin(
+    request: Drash.Http.Request,
+    response?: Drash.Http.Response,
+  ): void {
+    // If there is a response, then we know this is occurring after the request
+    if (response) {
+      // Set "X-XSS-Protection" header. See https://helmetjs.github.io/docs/xss-filter/
+      if (configs["X-XSS-Protection"] !== false) {
+        response.headers.set(
+          "X-XSS-Protection",
+          defaultConfigs["X-XSS-Protection"],
+        );
+        configs["X-XSS-Protection"] = true;
+      }
+
+      // Set "Referrer-Policy" header if passed in. See https://helmetjs.github.io/docs/referrer-policy/
+      if (configs["Referrer-Policy"]) {
+        response.headers.set(
+          "Referrer-Policy",
+          configs["Referrer-Policy"].toString(),
+        );
+      }
+
+      // Set the "X-Content-Type-Options" header. See https://helmetjs.github.io/docs/dont-sniff-mimetype/
+      if (configs["X-Content-Type-Options"] !== false) {
+        response.headers.set(
+          "X-Content-Type-Options",
+          defaultConfigs["X-Content-Type-Options"],
+        );
+        configs["X-Content-Type-Options"] = true;
+      }
+
+      // Set the "Strict-Transport-Security" header. See https://helmetjs.github.io/docs/hsts/
+      let hstsHeader = "";
+      if (configs.hsts!.max_age) { // if set to a number
+        hstsHeader += "max-age=" + configs.hsts!.max_age;
+      } else if (configs.hsts!.max_age !== false) { // not disabled
+        hstsHeader += "max-age=" + defaultConfigs.hsts.max_age;
+      }
+      if (hstsHeader && configs.hsts!.include_sub_domains === true) {
+        hstsHeader += "; include_sub_domains";
+      } else if (hstsHeader && configs.hsts!.include_sub_domains !== false) {
+        hstsHeader += "; include_sub_domains";
+      }
+      if (hstsHeader && configs.hsts!.preload === true) {
+        hstsHeader += "; preload";
+      }
+      // } else if (hstsHeader && configs.hsts!.preload !== false) {
```

Comment:
do we plan to use this? im assuming this is to check if it's any truthy value. is that correct?

Reply:
+1